test/tools/jsontests/TransactionTests.cpp

```diff
@@ -38,134 +39,167 @@ namespace fs = boost::filesystem;
+		bool onConstantinople = (network == eth::Network::ConstantinopleTest);
+		out[test::netIdToString(network)] = mObject();
+		mObject expectSection = getExpectSection(expectObj, network);
+		try
+		{
+			Transaction txFromFields(rlpStream.out(), CheckTransaction::Everything);
+			bool onConstantinopleAndZeroSig = onConstantinople && txFromFields.hasZeroSignature();
+			if (!(txFromFields.signature().isValid() || onConstantinopleAndZeroSig))
+				BOOST_THROW_EXCEPTION(Exception() << errinfo_comment(TestOutputHelper::get().testName() + "transaction from RLP signature is invalid") );
+
+			se->verifyTransaction(ImportRequirements::Everything, txFromFields, bh, 0);
+			if (expectSection.count("sender") > 0)
+			{
+				string expectSender = toString(expectSection["sender"].get_str());
+				BOOST_CHECK_MESSAGE(toString(txFromFields.sender()) == expectSender, "Error filling transaction test " + TestOutputHelper::get().testName() + ": expected another sender address! (got: " + toString(txFromFields.sender()) + "), expected: (" + expectSender + ")");
+			}
+
+			mObject resultObject;
+			resultObject["sender"] = toString(txFromFields.sender());
+			//resultObject["transaction"] = ImportTest::makeAllFieldsHex(tObj);
+			resultObject["hash"] = toString(txFromFields.sha3());
+			out[test::netIdToString(network)] = resultObject;
+		}
+		catch(Exception const& _e)
+		{
+			//Transaction is InValid
+			cnote << "Transaction Exception: " << diagnostic_information(_e);
+			bool expectInValid = (expectSection["result"].get_str() == "invalid");
+			BOOST_CHECK_MESSAGE(expectInValid, TestOutputHelper::get().testName() + " Check state: Transaction is expected to be valid! (" + test::netIdToString(network) + ")");
+			continue;
+		}
+
+		//Transaction is Valid
+		bool expectValid = (expectSection["result"].get_str() == "valid");
+		BOOST_CHECK_MESSAGE(expectValid, TestOutputHelper::get().testName() + " Check state: Transaction is expected to be invalid! (" + test::netIdToString(network) + ")");
+	}
+	return out;
+}
+
+void TestTransactionTest(json_spirit::mObject const& _o)
+{
+	BOOST_REQUIRE(_o.count("rlp") > 0);
+	Transaction txFromRlp;
+	string const& testname = TestOutputHelper::get().testName();
+
+	// Theoretical block for transaction check
+	BlockHeader bh;
+	bh.setNumber(1);	//Seal engine below enables network rules from block 0
+	bh.setGasLimit(u256("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
+
+	for (auto const network: test::getNetworks())
+	{
+		string networkname = test::netIdToString(network);
+		BOOST_REQUIRE_MESSAGE(_o.count(networkname) > 0, testname + " Transaction test missing network results! (" + networkname + ")");
+		BOOST_REQUIRE(_o.at(networkname).type() == json_spirit::obj_type);
+		mObject obj = _o.at(networkname).get_obj();
+		try
+		{
+			bytes stream = importByteArray(_o.at("rlp").get_str());
+			RLP rlp(stream);
+
+			txFromRlp = Transaction(rlp.data(), CheckTransaction::Everything);
```

Comment:
It's better to make this `Transaction txFromRlp = ...` and remove the definition above.

Reply:
this field is used after try, catch block.

Reply:
Then move the definition into the body of this `for` loop.

Reply:
done